src/event-bus/EventStore.ts

```diff
@@ -0,0 +1,106 @@
+import { Event, PersistedEvent } from '../shared/events';
+import { Logger } from '../shared/logger';
+import { promisify } from '../shared/promisify';
+import { sleep } from '../shared/sleep';
+
+const eventStoreConstructor = require('eventstore');
+
+export default class EventStore {
+
+  private isConnected = false;
+  private eventStore: any;
+
+  constructor(
+    private readonly publisher: (event: Event<any>, cb: (err: any) => void) => void,
+    private readonly eventStoreConfig: any,
+    private readonly logger: Logger,
+    private readonly numRetries = 1,
+    private readonly sleepFor = 300,
+  ) {
+  }
+
+  public async persistEvent(event: Event<any>) {
```

Comment:
It's a good idea to explicitly define the return value of the function. In addition to documenting the code, it'll catch bugs.